coveo/cli
# fix: handle invalid directory errors when creating atomic component

[KIT-2722](https://coveord.atlassian.net/browse/KIT-2722)

Unfortunately adding to the pile that is [CDX-1428](https://coveord.atlassian.net/browse/CDX-1428), but I couldn't find any simple/quick way to share the error class and behaviour between packages without doing larger changes.

## Proposed changes

It would be useful to provide a helpful error message when attempting to create an atomic component in an invalid directory. Thus, I added the necessary to the `create-atomic-component` and `create-atomic-result-component` packages.

## Testing

**UNIT TESTS ARE STILL IN PROGRESS**. For now, you can test the changes by following these steps:
- Build this branch (`npm run build`)
- Open a non-node project directory in your terminal (e.g., a java project)
- Try to create a component in said directory using your local build.
  - If the directory is _adjacent_ to where you have cloned the CLI project, your command will look something like
  ```
  ../cli/packages/cli/core/bin/dev atomic:cmp --type=result test-result-component
  ```
- Confirm that a clear error message outputs in your terminal, explaining that the current directory is invalid.

[KIT-2722]: https://coveord.atlassian.net/browse/KIT-2722?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CDX-1428]: https://coveord.atlassian.net/browse/CDX-1428?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
